File: src/test/java/io/tripled/social/client/infrastructure/InMemorySocialNetworkRepositoryTest.java
package io.tripled.social.client.infrastructure;

import io.tripled.social.client.domain.SocialNetwork;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.time.Clock;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.sameInstance;

class InMemorySocialNetworkRepositoryTest {

  private InMemorySocialNetworkRepository repository;

  @BeforeEach
  void setUp() {
    repository = new InMemorySocialNetworkRepository(new TestDateTimeProvider(Clock.systemDefaultZone()));
  }

  @Test
  void alwaysReturnsSameInstance() {
    SocialNetwork socialNetwork = repository.get();

    assertThat(socialNetwork, sameInstance(repository.get()));
  }
}

File: src/test/kotlin/io/tripled/social/client/infrastructure/InMemorySocialNetworkRepositoryTest.kt
package io.tripled.social.client.infrastructure

import io.tripled.social.client.domain.SocialNetwork
import org.hamcrest.MatcherAssert
import org.hamcrest.Matchers
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import java.time.Clock

internal class InMemorySocialNetworkRepositoryTest {
    private var repository: InMemorySocialNetworkRepository? = null

    @BeforeEach
    fun setUp() {
        repository = InMemorySocialNetworkRepository(TestDateTimeProvider(Clock.systemDefaultZone()))
    }

    @Test
    fun alwaysReturnsSameInstance() {
        val socialNetwork = repository!!.get()

        MatcherAssert.assertThat<SocialNetwork?>(
            socialNetwork,
            Matchers.sameInstance<SocialNetwork>(repository!!.get())
        )
    }
}